src/tests.rs

```diff
@@ -155,3 +155,168 @@ fn struct_with_generics() {
 		TypeDefStruct::new(vec![NamedField::new("data", struct_bool_id.clone()),]).into(),
 	);
 }
+
```

Comment:
We might also need tests that cover `RegisterSubtypes` implementations. However, I think this can be done in another PR that also tests the `Registry` as well.

Reply:
Agree, let's add register related tests in another PR.